Sources/SwiftKueryPostgreSQL/PostgreSQLConnection.swift

```diff
@@ -513,4 +542,46 @@ public class PostgreSQLConnection: Connection {
       }
       return postgresQuery
     }
+    
+    private func lockStateLock() {
+        _ = stateLock.wait(timeout: DispatchTime.distantFuture)
+    }
+    
+    private func unlockStateLock() {
+        stateLock.signal()
+    }
+    
+    func setState(_ newState: ConnectionState) {
+        lockStateLock()
+        if state == .fetchingResultSet {
+            currentResultFetcher = nil
+        }
+        state = newState
+        unlockStateLock()
+    }
+    
+    func setUpForRunningQuery() -> String? {
```

Comment:
This is weird, a setup function which returns an optional String?  Why not just make it `throw`?